phongo_compat.h

```diff
@@ -218,6 +218,15 @@
 	} while (0)
 #endif
 
+/* Per https://wiki.php.net/rfc/internal_method_return_types, "Non-final
+ * internal method return types - when possible - are declared tentatively in
+ * PHP 8.1, and they will become enforced in PHP 9.0." This can be revisited
+ * when more general typing improvements are made in PHPC-1709. */
+#ifndef ZEND_BEGIN_ARG_WITH_TENTATIVE_RETURN_TYPE_INFO_EX
+#define ZEND_BEGIN_ARG_WITH_TENTATIVE_RETURN_TYPE_INFO_EX(name, return_reference, required_num_args, type, allow_null) \
+	ZEND_BEGIN_ARG_INFO_EX(name, 0, return_reference, required_num_args)
+#endif
```

Comment:
This macro was introduced in https://github.com/php/php-src/commit/532c60cb921757151f4a60b9f56e2805ce23d819 (8.1.0alpha1). For earlier versions where this isn't required, we'll just fall back to standard arginfo.

Actual return type arginfo will be handled by [PHPC-1709](https://jira.mongodb.org/browse/PHPC-1709) down the line.

Reply:
Ah, this is nice so we don't have to add conditionals all over the place. Thanks!